test/quic_test.go

```diff
@@ -117,6 +121,38 @@ func TestQUICProtocolError(t *testing.T) {
 	}
 }
 
+func TestQUICPlugin(t *testing.T) {
```

Comment:
The plugin integration test ended up in the `test` folder. Unlike `forward`, setting up a mock `quic` server to answer a simple DNS query is not easy because you need not only to create a listener but also to accept connections, and streams, and finally write to the stream.

The best way I found was to use the `CoreDNSServerAndPorts` function to spin up a QUIC server and test the plugin against it.